src/main/java/com/example/bookstore/book/BookService.java

```diff
@@ -27,6 +27,25 @@ public Iterable<Book> getBookByQuery(String query) {
 
     }
 
+    public Iterable<Book> sortBooksByQuery(String query) {
+        switch (query) {
+            case "title" -> {
+                return this.bookRepository.findAllByOrderByTitleAsc();
+            }
+            case "author" -> {
+                return this.bookRepository.findAllByOrderByAuthorAsc();
+            }
+            case "publisher" -> {
+                return this.bookRepository.findAllByOrderByPublisherAsc();
+            }
+            case "stock" -> {
+                return this.bookRepository.findAllByOrderByStockAsc();
+            }
+        }
+        // somehow invalid sort param passed
+        return null;

```

Comment:
Should we be throwing an exception here instead?

Reply:
not sure. I think null works as well